app/models/plate.rb

```diff
@@ -146,10 +154,14 @@ def source_plate
       ]
     )
   }
+  scope :with_plate_purpose, ->(*purposes) { where(plate_purpose_id: purposes.flatten) }
 
   # About 10x faster than going through the wells
   def submission_ids
     @siat ||= container_associations
+              .joins('LEFT JOIN transfer_requests ON transfer_requests.target_asset_id = container_associations.content_id')
```

Comment:
I'm not sure this actually needs to be this complicated. This maintains existing behaviour, but we might be able to get away with something simpler.